### PR TITLE
Change CreateAPICall to Invoke

### PR DESCRIPTION
**Pull request by saicheems**
Allows the user to pass a function which takes a modified context based
on the provided call options.
The benefit is avoiding runtime casting in the generated surface.

### CODE REVIEW

**Comment by saicheems** [comment]
@jmuk 
@vchudnov-g 
@jba 
@pongad

File: api_callable.go [hunk 1]
```diff
@@ -8,68 +8,59 @@ import (
...
+		// If the deadline is exceeded...
+		if childCtx.Err() != nil {
+			return childCtx.Err()
+		}
+		timeoutCtx, _ := context.WithTimeout(childCtx, backoffSettings.rpcTimeoutSettings.max)
+		timeoutCtx, _ = context.WithTimeout(timeoutCtx, timeout)
+		err = stub(timeoutCtx)
+		code := grpc.Code(err)
+		if code == codes.OK {
+			return
```
Comment (jmuk) [inline api_callable.go]: You might want to specify the return value (`return nil`) here. According to https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters,

> Naked returns are okay if the function is a handful of lines.

Comment (jba) [inline api_callable.go]: Agreed, try to avoid naked returns. I also wouldn't name the return value, to avoid temptation/confusion.

Comment (saicheems) [inline api_callable.go]: Makes sense, fixed

**Comment by jmuk** [comment]
LGTM

**Pull request merged**